acceptance-tests/src/test/java/it/com/atlassian/bitbucket/jenkins/internal/SmokeTest.java

```diff
@@ -0,0 +1,444 @@
+
+@WithPlugins({"mailer", "atlassian-bitbucket-server-integration"})
+public class SmokeTest extends AbstractJUnitTest {
+
+    @Rule
+    public final TemporaryFolder tempFolder = new TemporaryFolder();
+
+    private String bbsAdminCredsId;
+    private BitbucketRepository forkRepo;
+    private String serverId;
+
+    @Before
+    public void setUp() {
+        // BB Access Token
+        PersonalToken bbsToken = createPersonalToken(PROJECT_READ_PERMISSION, REPO_ADMIN_PERMISSION);
+        CredentialsPage credentials = new CredentialsPage(jenkins, DEFAULT_DOMAIN);
+        credentials.open();
+        BitbucketTokenCredentials bbsTokenCreds = credentials.add(BitbucketTokenCredentials.class);
+        bbsTokenCreds.token.set(bbsToken.getSecret());
+        bbsTokenCreds.description.sendKeys(bbsToken.getId());
+        credentials.create();
+
+        // BB Admin user/password
+        credentials.open();
+        UserPwdCredential bbAdminCreds = credentials.add(UserPwdCredential.class);
+        bbsAdminCredsId = "admin-" + randomUUID();
+        bbAdminCreds.setId(bbsAdminCredsId);
+        bbAdminCreds.username.set(BITBUCKET_ADMIN_USERNAME);
+        bbAdminCreds.password.set(BITBUCKET_ADMIN_PASSWORD);
+        credentials.create();
+
+        // Create BB Server entry
+        JenkinsConfig jenkinsConfig = jenkins.getConfigPage();
+        jenkinsConfig.configure();
+        serverId = "bbs-" + randomUUID();
+        new BitbucketPluginConfiguration(jenkinsConfig)
+                .addBitbucketServer(serverId, BITBUCKET_BASE_URL, bbsToken.getId(), bbsAdminCredsId);
+        jenkinsConfig.save();
+
+        // Fork repo
+        forkRepo = forkRepository(PROJECT_KEY, REPO_SLUG, "fork-" + randomUUID());
+    }
+
+    @Test
+    public void testFreeStyleJob() throws IOException, GitAPIException {
+        FreeStyleJob freeStyleJob = jenkins.jobs.create();
+        BitbucketScm bitbucketScm = freeStyleJob.useScm(BitbucketScm.class);
+        bitbucketScm
+                .credentialsId(bbsAdminCredsId)
+                .serverId(serverId)
+                .projectName(forkRepo.getProject().getKey())
+                .repositoryName(forkRepo.getSlug())
+                .anyBranch();
+        freeStyleJob.addTrigger(BitbucketWebhookTrigger.class);
+        freeStyleJob.save();
+
+        // Clone (fork) repo and push new file
+        CredentialsProvider cr =
+                new UsernamePasswordCredentialsProvider(BITBUCKET_ADMIN_USERNAME, BITBUCKET_ADMIN_PASSWORD);
+        File checkoutDir = tempFolder.newFolder("repositoryCheckout");
+        Git gitRepo = cloneGitRepo(cr, checkoutDir, forkRepo);
+
+        final String branchName = "smoke/test";
+        gitRepo.branchCreate().setName(branchName).call();
+        gitRepo.checkout().setName(branchName).call();
+        File jenkinsFile = new File(checkoutDir, "Jenkinsfile");
+        try (InputStream in = getClass().getResourceAsStream("/sampleJenkinsfile")) {
```

Comment:
We should look into inlining these jenkins file content in the test to enhance clarity. These files are small and they convey what the test is doing. For example, 
![image](https://user-images.githubusercontent.com/7725996/87548473-df0ee480-c6ef-11ea-850d-275a4499cfd1.png)

Reply:
There's no "like" in GH but I considered saying the same thing... so this gets a +1 from me.

Reply:
> Isn't that just noise though?

I think we all have different ideas of what's noise. 😛 

For me, literally the first thing I wanted to know was "what does this buildfile do?" and it was a _minor_ inconvenience to have to go to another file instead of have it as a constant (which, is how I'd do it, instead of having it inlined in the test - since it's the same for all tests).

But I really don't feel super strongly about it.

Reply:
> For me, literally the first thing I wanted to know was "what does this buildfile do

Why though? Because as an SCM we really never care about that, right? The contents of the file is sort of irrelevant for the test, we're not testing parsing of it, or anything. We just care that it runs successfully; so I prefer to have it in a file, as it is just part of the test setup. Just like you don't care what password is used to authenticate or anything such, as long as it works?

I'm happy with either solution though, not overly fussed.

Reply:
> Because as an SCM we really never care about that, right?
Incorrect. Jenkins file can contain SCM related information. I am happy if JenkinsFile is part of a string constant but accessing it through IDE is much easier than searching for a file in the resource. Especially when the size of it is 4 lines only. 
Inlining becomes more useful when you have different cases of Jenkins file handled in tests. For example, 

1. Groovy script 
![image](https://user-images.githubusercontent.com/7725996/87642042-542ff780-c78c-11ea-8bfb-e9a368ef0cb3.png)
2. Bitbucket SCM in pipeline script, 
![image](https://user-images.githubusercontent.com/7725996/87642132-74f84d00-c78c-11ea-8814-4f0a8b31e8a1.png)

Reply:
- [ ] move Jenkinsfile content to a constant instead